src/coreclr/jit/fginline.cpp

```diff
@@ -586,8 +606,51 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
                 }
 
                 CORINFO_CONTEXT_HANDLE contextInput = context;
+                context                             = nullptr;
                 m_compiler->impDevirtualizeCall(call, nullptr, &method, &methodFlags, &contextInput, &context,
                                                 isLateDevirtualization, explicitTailCall);
+                // TODO-CQ: We should spill the call if it has side effects instead of conservatively
+                // estimating the side effects using its parent and blocking inlining.
+                if (context != nullptr && (parent == nullptr || parent->OperIs(GT_STORE_LCL_VAR)))
```

Comment:
```suggestion
                if (context != nullptr && (parent == nullptr || (parent->OperIs(GT_STORE_LCL_VAR) && (parent == m_curStmt->GetRootNode()))))
```
If there is store parent, verify it is the statement root (to rule out the store being part of a comma).